golang/src/dockerops/containers.go

```diff
@@ -159,6 +168,26 @@ func (d *Docker) NukeImage(name, tag string) error {
 	return d.Client.RemoveImageExtended(imageName, opts)
 }
 
+// Images will returns a list of the repo tags for all the images currently
+// downloaded.
+func (d *Docker) Images() ([]string, error) {
+	opts := docker.ListImagesOptions{
+		All: true,
+	}
+	apiImages, err := d.Client.ListImages(opts)
+	if err != nil {
+		return nil, err
+	}
+	var retval []string
+	for _, img := range apiImages {
+		repos := img.RepoTags
```

Comment:
If I'm not mistaken, this will mean that certain situations will leave around old images or break this logic. Specifically:

* image A:latest is used by a job
* A:latest is pushed with something new (this happens at least with curl-wrapper and backwards-compat, if not always frequently)
* another job uses A:latest, pulling the new image
* the old image that was previously A:latest is now without any RepoTags

This might be handled by the existing docker cleanup of dangling images that we do, but the ansible stuff for that is currently broken and we haven't really been maintaining it (which means it's not on newly-provisioned or reprovisioned nodes), so we might want to get it right here.